### PR TITLE
TF-768: [Tablet/Mobile] Share image with TeamMail app

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/182308220-9d25513b-ece8-404b-8a7e-27636376cc67.mov

https://user-images.githubusercontent.com/107173849/182308364-d9f6dcbf-1554-41cb-8c31-443a81e54a04.mov

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -144,7 +144,10 @@ dependencies:
 
   package_info_plus: 1.4.1
 
-  receive_sharing_intent: 1.4.5
+  receive_sharing_intent:
+    git:
+      url: https://github.com/ManhNTX/receive_sharing_intent.git
```
Comment (dab246) [inline pubspec.yaml:149]: Describe more clearly what you changed the library in this PR.

**Review by dab246 — COMMENTED** [review]
IMO, we only need `Share Image` now so you don't need to add handler functions for `video` and `file`

File: android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+import android.os.PersistableBundle
 import io.flutter.embedding.android.FlutterFragmentActivity
 import io.flutter.embedding.engine.FlutterEngine
 import io.flutter.plugins.GeneratedPluginRegistrant
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
 
 class MainActivity: FlutterFragmentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
+        if (intent.getIntExtra("org.chromium.chrome.extra.TASK_ID", -1) == this.taskId) {
```
Comment (hoangdat) [inline android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt:13]: what this use for?

File: android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -69,6 +69,40 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.SEND" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="video/*" />
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND_MULTIPLE" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="video/*" />
+            </intent-filter>
```
Comment (hoangdat) [inline android/app/src/main/AndroidManifest.xml]: you forgot sth?

File: android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt (hunk 2 above)
Comment (hoangdat) [reply android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt:13]: ??

File: ios/TeamMailShareExtension/Info.plist [hunk 4]
```diff
@@ -26,16 +26,27 @@
 		<dict>
 			<key>NSExtensionActivationRule</key>
 			<dict>
+				<key>NSExtensionActivationSupportsFileWithMaxCount</key>
+				<integer>15</integer>
+				<key>NSExtensionActivationSupportsImageWithMaxCount</key>
+				<integer>15</integer>
+				<key>NSExtensionActivationSupportsMovieWithMaxCount</key>
```
Comment (hoangdat) [inline ios/TeamMailShareExtension/Info.plist:33]: not support movie ATM

**Review by hoangdat — COMMENTED** [review]
Please address what you change for iOS controller in an `ADR`. And please don't forget to reply comments.

**Comment by hoangdat** [comment]
Please keep the `Share Videos` implementation for another ticket.

File: pubspec.yaml (hunk 1 above)
Comment (ManhNTX) [reply pubspec.yaml:149]: I change the app group id like below 
Screen Shot 2022-08-03 at 06 40 29

File: android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt (hunk 2 above)
Comment (ManhNTX) [reply android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt:13]: https://pub.dev/packages/receive_sharing_intent 
follow document

**Comment by ManhNTX** [comment]
> Please keep the `Share Videos` implementation for another ticket.

I will add later if has ticket

File: pubspec.yaml (hunk 1 above)
Comment (hoangdat) [reply pubspec.yaml:149]: We have any way to do it in compile time?

File: android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt (hunk 2 above)
Comment (ManhNTX) [reply android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt:13]: When I'm developing an app that receives data using intents(ACTION_SEND). I've discovered a weird issue when I'm sharing content from Chrome(only happens in Chrome 83 or above, older versions can't reproduce the issue) for instance, if the content is only selected text, for example, the data reaches my app correctly, but if the content shared is a URL, it reaches my app, but suddenly my app replaces Chrome in the app switcher. So, if I have my app and Chrome opened, after the data sharing I have two instances of my app(even if the Chrome icon appears in top of that window, if I tap it, opens my app).

This code to resolve this problem

File: pubspec.yaml (hunk 1 above)
Comment (ManhNTX) [reply pubspec.yaml:149]: > Please address what you change for iOS controller in an `ADR`. And please don't forget to reply comments.

Done

**Comment by ManhNTX** [comment]
> Please address what you change for iOS controller in an `ADR`. And please don't forget to reply comments.

Done

File: lib/features/composer/presentation/composer_controller.dart [hunk 5]
```diff
@@ -900,7 +902,42 @@ class ComposerController extends BaseController {
     }
   }
 
+  List<InlineImage> covertListSharedMediaFileToFile(List<SharedMediaFile> value) {
+    List<File> newFiles = List.empty(growable: true);
+    if (value.isNotEmpty) {
+      for (var element in value) {
+        newFiles.add(File(
+          Platform.isIOS
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:910]: Should be written more clearly. Ex:
```dart
Platform.isIOS
   ? element.type == SharedMediaType.FILE
        ? element.path.toString().replaceAll('file:/', '')
        : element.path
   : element.path,
```

File: ios/TeamMailShareExtension/Info.plist [hunk 6]
```diff
@@ -26,16 +26,23 @@
...
-                <true/>
-                <key>NSExtensionActivationSupportsWebURLWithMaxCount</key>
-                <integer>1</integer>
+				<true/>
+				<key>NSExtensionActivationSupportsWebURLWithMaxCount</key>
+				<integer>1</integer>
 			</dict>
+			<key>PHSupportedMediaTypes</key>
+			<array>
+				<string>Video</string>
```
Comment (dab246) [inline ios/TeamMailShareExtension/Info.plist:38]: remove it

File: pubspec.yaml (hunk 1 above)
Comment (ManhNTX) [reply pubspec.yaml:149]: @hoangdat  done

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]

**Pull request merged**